bin/git-bulk

```diff
@@ -36,7 +36,11 @@ function addworkspace {
             source=$(realpath "$source" 2>/dev/null)
             if [ -f "$source" ]; then
                 pushd "$wsdir" > /dev/null
-                while read -r line; do git clone "$line"; done < "$source";
+                while IFS= read -r line || [[ -n $line ]]; do
```

Comment:
Do we need to use `-n "$line"` here?

Reply:
`-n "$line"` checks whether the variable $line is non-empty. It's needed since the `repository.txt` file can contain empty line.

Reply:
Yes, you are right, we don't need `-n $line`. I have updated the code.